src/main/java/edu/phema/elm_to_omop/vocabulary/MultiModalValuesetService.java

```diff
@@ -0,0 +1,57 @@
+package edu.phema.elm_to_omop.vocabulary;
+
+import edu.phema.elm_to_omop.vocabulary.phema.PhemaConceptSet;
+import edu.phema.elm_to_omop.vocabulary.phema.PhemaConceptSetList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MultiModalValuesetService implements IValuesetService {
+    private List<IValuesetService> valuesetServices;
+
+    private int codesetId;
+
+    public MultiModalValuesetService(IValuesetService... valuesetServices) {
+        this.valuesetServices = new ArrayList<>();
+
+        for (IValuesetService valuesetService : valuesetServices) {
+            this.valuesetServices.add(valuesetService);
+        }
+
+        codesetId = 0;
+    }
+
+    private PhemaConceptSetList mergeConceptSetLists(List<PhemaConceptSetList> phemaConceptSetLists) {
+        PhemaConceptSetList phemaConceptSetListResult = new PhemaConceptSetList();
+
+        for (PhemaConceptSetList phemaConceptSetList : phemaConceptSetLists) {
+            List<PhemaConceptSet> conceptSets = phemaConceptSetList.getConceptSets();
+
+            // Update IDs so we don't get duplicates
+            for (PhemaConceptSet phemaConceptSet : conceptSets) {
+                phemaConceptSet.id = codesetId++;
+            }
+            phemaConceptSetListResult.addAllConceptSets(conceptSets);
+
+            phemaConceptSetListResult.addAllNotFoundCodes(phemaConceptSetList.getNotFoundCodes());
+        }
+
+        return phemaConceptSetListResult;
+    }
+
+    @Override
+    public PhemaConceptSetList getConceptSetList() throws ValuesetServiceException {
+        List<PhemaConceptSetList> phemaConceptSetLists = new ArrayList<>();
+
+        for (IValuesetService valuesetService : valuesetServices) {
+            phemaConceptSetLists.add(valuesetService.getConceptSetList());
+        }
+
+        return mergeConceptSetLists(phemaConceptSetLists);
```

Comment:
This may be intended (or acceptable) behavior - but if I call `getConceptSetList` or `getConceptSets` multiple times, it looks like it will re-merge the lists and would reassign new IDs.  If this is potentially an issue (may not be depending on how the class is used), we could cache the merged list.

Reply:
This is a bug, nice catch. I'll fix it, and also probably implement in-memory caching so we don't even have to merge the lists (and potentially call out to the server) again.